src/Microsoft.AspNet.Http/SafeDictionary.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Http.Internal
+{
+    public class SafeDictionary<TKey, TValue> : IDictionary<TKey, TValue>
+    {
+        public IDictionary<TKey, TValue> Store { get; set; }
+
+        public SafeDictionary(IDictionary<TKey, TValue> store)
+        {
+            Store = store;
+        }
+
+        public TValue this[TKey key]
+        {
+            get
+            {
+                if (Store == null)
+                {
+                    new NullReferenceException();
```

Comment:
Its a pass through wrapper that's only purpose it to massage ```Get```. If Store is not set that's the equivalent of not having the wrapper and not setting the original collection to anything.

You probably want to know this error up front, and the behaviour maintained? Rather than the disconnect disappearing into the void and just producing non-nonsensical results.

Ideally you wouldn't need class at all for this particular use.